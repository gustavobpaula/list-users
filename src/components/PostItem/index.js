import React from 'react';
import PropTypes from 'prop-types';
import {
  Post,
  User,
  BoxTitle,
  Title,
  BoxDescription,
  Description,
} from './style';


export default function PostItem({
  title,
  body,
  user,
  index,
}) {
  const userData = user.data;

  return (
    <Post>
      <BoxTitle className={index % 2 === 0 ? 'even' : ''}>
        <Title>{title}</Title>
        <User>
          {userData && userData.name} | {userData && userData.company && userData.company.name}
        </User>
      </BoxTitle>
      <BoxDescription className={index % 2 === 0 ? 'even' : ''}>
        <Description>{body}</Description>
      </BoxDescription>
    </Post>
  );
}

PostItem.propTypes = {
  index: PropTypes.number.isRequired,
  title: PropTypes.string.isRequired,
  body: PropTypes.string.isRequired,
  user: PropTypes.shape({
    data: PropTypes.shape({
      name: PropTypes.string.isRequired,
      company: PropTypes.shape({
        name: PropTypes.string.isRequired,
      }).isRequired,
    }),
  }),
};
