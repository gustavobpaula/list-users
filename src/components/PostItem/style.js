import styled from 'styled-components';

export const Post = styled.article`
  border-bottom: 1px solid #ccc;
  display: grid;
  grid-gap: 50px;
  grid-template-columns: 1fr 1fr;
  padding: 80px 0;

  &:nth-child(even) {
    grid-auto-flow: dense
  }
`;

export const BoxTitle = styled.div`
  grid-column: 2;

  &.even {
    grid-column: 1;
  }
`;

export const BoxDescription = styled.div`
  grid-column: 1;

  &.even {
    grid-column: 2;
  }
`;

export const Title = styled.h1`
  color: #333;
  font-size: 45px;
  margin: 0;

  @media (max-width: 425px) {
    font-size: 25px;
  }
`;

export const User = styled.p`
  color: #999;
`;

export const Description = styled.p`
  line-height: 200%;
  margin: 0;
`;
