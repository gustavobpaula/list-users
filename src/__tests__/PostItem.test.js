import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import PostItem from '../components/PostItem';

describe('Testing PostItem Component', () => {
  it('should render correctly', () => {
    const props = {
      title: 'Lorem Ipsum',
      body: `Lorem ipsum dolor sit amet,
          consectetur adipiscing elit. Suspendisse lacus dui, varius vitae ligula quis, suscipit dapibus neque.
          Curabitur aliquam orci lacus, nec ultricies dui rhoncus nec. Ut in orci sit amet libero tincidunt euismod.
          Vivamus sodales ut neque eu dapibus. Nulla metus justo, molestie nec tellus sed, sollicitudin rhoncus.`,
      user: {
        data: {
          name: 'Lorem Ipsum',
          company: {
            name: 'Lorem Ipsum',
          },
        },
      },
      index: 1,
    };

    const wrapper = shallow(
      <PostItem
        index={props.index}
        title={props.title}
        body={props.body}
        user={props.user}
      />,
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
