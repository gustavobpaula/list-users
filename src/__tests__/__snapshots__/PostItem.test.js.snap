// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing PostItem Component should render correctly 1`] = `
<styled.article>
  <styled.div
    className=""
  >
    <styled.h2>
      Lorem Ipsum
    </styled.h2>
    <styled.p>
      Lorem Ipsum
       | 
      Lorem Ipsum
    </styled.p>
  </styled.div>
  <styled.div
    className=""
  >
    <styled.p>
      Lorem ipsum dolor sit amet,
          consectetur adipiscing elit. Suspendisse lacus dui, varius vitae ligula quis, suscipit dapibus neque.
          Curabitur aliquam orci lacus, nec ultricies dui rhoncus nec. Ut in orci sit amet libero tincidunt euismod.
          Vivamus sodales ut neque eu dapibus. Nulla metus justo, molestie nec tellus sed, sollicitudin rhoncus.
    </styled.p>
  </styled.div>
</styled.article>
`;
